Add EIP: Universal Enshrined Encrypted Mempool

jannikluhn opened the pull request:
This PR adds an EIP to introduce a scheme agnostic encrypted mempool to Ethereum.

jannikluhn committed:
feat: add UEEM EIP

abcoathup commented on EIPS/eip-ueem.md:
```diff
@@ -0,0 +1,116 @@
+---
+title: Universal Enshrined Encrypted Mempool
```
```suggestion
title: Encrypted Transaction Pool
```
Note: I am not an editor.

Universal isn't used anywhere else in the EIP.
Enshrined is implied as an EIP.
Ethereum has transaction pools.

jannikluhn committed:
chore: use assigned EIP number

g11tech reviewed (commented):
pls get the CI to pass so that this can be reviwed

jannikluhn committed:
fix: remove external links

g11tech commented on EIPS/eip-8105.md:
```diff
@@ -0,0 +1,117 @@
+
+## Specification
+
+In the execution layer, a contract called the key provider registry is deployed. It allows any account to register a key provider and assigns them a unique ID. Registration requires specifying a contract with a decryption and a key validation function, each of which accept a key ID and a key message as byte strings. Additionally, key providers may designate other providers as directly trusted, th ...
```
is this a system contract? i.e. recognized by the clients, if thats the case then it would be better to call it a system contract

jannikluhn committed:
feat/fix: clarifications and fixes

- improve structure
- switch to explicit 2 slot design
- specify transaction formats
- discuss payload/envelope stripping attacks
- add key provider registry contract code placeholder
- add links to related EIPs

g11tech commented on EIPS/eip-8105.md:
```diff
@@ -0,0 +1,153 @@
+---
+eip: 8105
+title: Universal Enshrined Encrypted Mempool
```
```suggestion
title: Universally Enshrined Encrypted Mempool
```

g11tech reviewed (commented):
should have security considerations section
also recommend to address how "Free DA" problem is resolved or not resolved

lightclient reviewed (approved):
LGTM
